packages/typescript-estree/src/convert.ts

```diff
@@ -379,14 +379,27 @@ export class Converter {
    * property instead of a kind property. Recursively copies all children.
    */
   private deeplyCopy(node: TSNode): any {
+    if (
+      node.kind >= SyntaxKind.FirstJSDocNode &&
+      node.kind <= SyntaxKind.LastJSDocNode
+    ) {
+      throw createError(
+        this.ast,
+        node.pos,
+        'JSDoc types can only be used inside documentation comments.',
+      );
+    }
```

Comment:
don't we have a mechanism to bubble up compiler errors?
It seems weird that we have to duplicate the ts error logic here instead of just adding an error code.

Reply:
"bubble up compiler errors" is hidden behind flag, and its not enabled by default